### PR TITLE
Add panic recovery with error messages in L4Controller

### PR DESCRIPTION
**Pull request by FelipeYepez**
- Add panic recovery to syncWrapper, AddFunc, and UpdateFunc
- Log detailed error messages
- Emit metric on panic for monitoring

/assign @mmamczur @cezarygerard

### CODE REVIEW

**Comment by aojea** [comment]
why there should be a panic in an event handler?

File: pkg/l4lb/l4controller.go [hunk 1]
```diff
@@ -120,8 +120,15 @@ func NewILBController(ctx *context.ControllerContext, stopCh <-chan struct{}, lo
 		AddFunc: func(obj interface{}) {
 			addSvc := obj.(*v1.Service)
 			svcKey := utils.ServiceKeyFunc(addSvc.Namespace, addSvc.Name)
-			needsILB, svcType := annotations.WantsL4ILB(addSvc)
 			svcLogger := logger.WithValues("serviceKey", svcKey)
+			defer func() {
```
Comment (aojea) [inline pkg/l4lb/l4controller.go:124]: @cezarygerard talked with @FelipeYepez offline, I don't think we should use recovers on handlers and be paranoid, handlers must only do basic filtering and enqueu the event

/hold

File: pkg/l4lb/l4controller.go [hunk 2]
```diff
@@ -411,7 +425,17 @@ func (l4c *L4Controller) linkNEG(l4 *loadbalancers.L4, svcLogger klog.Logger) er
 func (l4c *L4Controller) syncWrapper(key string) error {
 	syncTrackingId := rand.Int31()
 	svcLogger := l4c.logger.WithValues("serviceKey", key, "syncId", syncTrackingId)
-	return skipUserError(l4c.sync(key, svcLogger), svcLogger)
+
+	var syncErr error
+	defer func() {
+		if r := recover(); r != nil {
+			errMessage := fmt.Sprintf("Panic in L4 ILB sync worker goroutine: %v", r)
```
Comment (aojea) [inline pkg/l4lb/l4controller.go:432]: I do not object to have this here, though , I don't think is a good practice

Comment (FelipeYepez) [reply pkg/l4lb/l4controller.go:418]: Thank you Antonio, after our conversation I'll leave the recover function for the syncWrapper.

File: pkg/l4lb/l4controller.go (hunk 1 above)
Comment (FelipeYepez) [reply pkg/l4lb/l4controller.go]: Thank you Antonio for your insights, removed this recover functions from here

**Comment by aojea** [comment]
/ok-to-test
/hold cancel
/lgtm

Thanks

File: pkg/l4lb/l4controller.go [hunk 3]
```diff
@@ -411,7 +411,17 @@ func (l4c *L4Controller) linkNEG(l4 *loadbalancers.L4, svcLogger klog.Logger) er
...
+			svcLogger.Error(nil, errMessage)
+			l4metrics.PublishL4ControllerPanicCount(l4ILBControllerName)
+		}
+	}()
+	syncErr = l4c.sync(key, svcLogger)
+	return skipUserError(syncErr, svcLogger)
```
Comment (mmamczur) [inline pkg/l4lb/l4controller.go:424]: as discussed offline, do we know what the syncWrapper will return if recover happens?

**Comment by aojea** [comment]
/hold

there are some comments form Michal, also I didn't approve it 🤔 , I was just waiting for someone else to do it

File: pkg/l4lb/l4controller.go (hunk 3 above)
Comment (FelipeYepez) [reply pkg/l4lb/l4controller.go:424]: Thank you Michal. The recover function will now return an error that will requeue the task.

**Pull request merged**